lib/SimplePie/SimplePie.php

```diff
@@ -1322,12 +1322,21 @@ public function enable_exceptions($enable = true)
 
 	function cleanMd5($rss)
 	{
-		return md5(preg_replace(array(
-			'#<(lastBuildDate|pubDate|updated|feedDate|dc:date|slash:comments)>[^<]+</\\1>#',
-			'#<(media:starRating|media:statistics) [^/<>]+/>#',
-			'#<!--.+?-->#s',
-			), '', $rss));
-		
+		$data = '';
+
+		$stream = fopen('php://memory','r+');                             
+		fwrite($stream, $rss);                                           
+		rewind($stream);                                                  
+																		  
+		while (($stream_data = fread($stream, 16384))) {                  
+			$data .= preg_replace(array(
```

Comment:
What happens if the target of the regular expression happens to be cut in two?

Reply:
@Kiblyn11 In your tests, was this part also crashing, or only the one below regarding `xml_parse()` ?

Reply:
@Frenzie you're right, that case fails.
@Alkarex Yes it was crashing, preg_replace is a memory hog when used with complex regex on fat content

I guess splitting isn't ideal here for that reason, I'm not sure what is the purpose of this cleaning, maybe there is a less expensive way to do it.

Reply:
The reason is detailed in https://github.com/FreshRSS/FreshRSS/commit/9aab83af115de3b210ea41caae49b70a0f82492a so it looks fairly important to indeed filter them out.

I think https://www.php.net/manual/en/book.xmlreader.php should be able to handle this better (on a per-node basis). It's detailed [here](https://drib.tech/programming/parse-large-xml-files-php).

NB Just brainstorming. I don't have personal experience with it.

Reply:
Ideally, the filtering should be done after the XML parsing, if not too slow. It is not a big problem if some deletions are not done, as it will only increase the risk slightly of not being able to cache the feed once.